File: common/src/main/java/cc/turtl/chiselmon/command/InfoCommand.java
package cc.turtl.chiselmon.command;

import cc.turtl.chiselmon.ChiselmonConstants;
import cc.turtl.chiselmon.util.MessageUtils;
import com.mojang.brigadier.Command;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.context.CommandContext;
import net.minecraft.client.Minecraft;
import net.minecraft.client.player.LocalPlayer;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;

public class InfoCommand implements ChiselmonCommand {

    @Override
    public String getName() {
        return "info";
    }

    @Override
    public String getDescription() {
        return "Display mod info";
    }

    @Override
    public LiteralArgumentBuilder<CommandSourceStack> build() {
        return Commands.literal(getName())
                .executes(this::execute);
    }

    private int execute(CommandContext<CommandSourceStack> context) {
        LocalPlayer player = Minecraft.getInstance().player;
        if (player == null) return 0;

        MessageUtils.sendHeader(player, ChiselmonConstants.MOD_NAME + " Info");
        MessageUtils.sendLabeled(player, "Version", ChiselmonConstants.VERSION);
        MessageUtils.sendLabeled(player, "Author", ChiselmonConstants.AUTHOR);
        return Command.SINGLE_SUCCESS;
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/command/InfoCommand.kt
package cc.turtl.chiselmon.client.command

import cc.turtl.chiselmon.BuildDetails
import cc.turtl.chiselmon.client.util.sendEmptyLine
import cc.turtl.chiselmon.client.util.sendLabeled
import cc.turtl.chiselmon.client.util.sendSuccess
import cc.turtl.turtlshell.api.core.command.TurtlShellCommand
import com.mojang.brigadier.Command
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.minecraft.client.Minecraft
import net.minecraft.commands.CommandSourceStack
import net.minecraft.network.chat.Component
import net.minecraft.network.chat.MutableComponent

class InfoCommand : TurtlShellCommand {

    override val name = "info"
    override val description: MutableComponent = Component.literal("Display mod info")

    override fun build(): LiteralArgumentBuilder<CommandSourceStack> =
        LiteralArgumentBuilder.literal<CommandSourceStack>(name)
            .executes {
                val player = Minecraft.getInstance().player ?: return@executes 0
                sendEmptyLine(player)
                sendSuccess(player, "${BuildDetails.MOD_DISPLAY_NAME} Info")
                sendLabeled(player, "  Version", BuildDetails.MOD_VERSION)
                sendLabeled(player, "  Author", BuildDetails.MOD_AUTHOR)
                Command.SINGLE_SUCCESS
            }
}